### PR TITLE
Update/fix Flatpak AppStream CI steps

### PR DESCRIPTION
**Pull request by tytan652**
### Description

TLDR: Replace use of appstream-glib tooling with flatpak-builder-lint + libappstream (unfortunately a fork)

Sorry, the not TLDR would end up in a rant.
But I can say that AppStream validation is also added to push/PR workflow to catch issues before release.

### Motivation and Context

Make Flatpak release finally work…

### How Has This Been Tested?

CI on my fork and this PR

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by GeorgesStavracas — APPROVED** [review]
I had an almost exactly line-by-line equal change, but you beat me up on publishing it :)

LGTM
